### PR TITLE
Fix/bugsandimprovements

### PR DESCRIPTION
**Pull request by pieryvdl**
Hi guys, small pull request from me, 

1. I updated the stats on the profile page to look a bit better. Note this is not the logic of the stats, this is purely frontend wise. 
2. I also moved te files into components and updated GameyCon to have a level 3. 
3. Fixed the display of the score and rating and added a star icon, I can change it to include only stars instead of the rating, but i thought the way it looks is clean.

Let me know what you guys think!

### CODE REVIEW

**Review by MonikaTheiss — APPROVED** [review]
Looks awesome ! Thank you 😄

**Review by shan-226 — APPROVED** [review]
Looks really good.

**Review by u22520016 — CHANGES_REQUESTED** [review]
I like all the changes/improvements you made.  I just have a comment on the ratings and score.  I think the padding between the text and the white border can increase.  I also do not really like the white border.  I would rather suggest making the background primaryContainer and removing the border.  I would also rather place these 2 things underneath the publisher of the game.  I would then also maybe change the score to look like the new one on the game library.  Almost like this:

Screenshot 2024-09-24 at 16 50 59

**Comment by pieryvdl** [comment]
@u22520016 Done as requested. ✅

File: lib/view/pages/game_library/game_details_page.dart
Comment (u22520016) [inline lib/view/pages/game_library/game_details_page.dart:1]: Please add the 12 px padding around the boxes and more padding inside and also between the two boxes...

**Review by u22520016 — CHANGES_REQUESTED** [review]

File: lib/view/pages/game_library/game_details_page.dart
Comment (pieryvdl) [reply lib/view/pages/game_library/game_details_page.dart:1]: Done.

**Review by u22520016 — APPROVED** [review]
That looks perfect, thank you.

**Pull request merged**